src/util-mpm-ac.c

```diff
@@ -131,10 +153,10 @@ static void SCACGetConfig()
 static inline int SCACReallocState(SCACCtx *ctx, uint32_t cnt)
 {
     void *ptmp;
-    int size = 0;
+    size_t size = 0;
 
     /* reallocate space in the goto table to include a new state */
-    size = cnt * ctx->single_state_size;
+    size = SCACCheckSafeSizetMult((size_t) cnt, (size_t) ctx->single_state_size);
     ptmp = SCRealloc(ctx->goto_table, size);
```

Comment:
Sure, I'll look into it. I can totally understand the goal of static checker cleanness -- we have also used scan-build (as a successor of splint) in previous projects.